apps/cli/src/node/docker-compose-espresso.yaml

```diff
@@ -0,0 +1,71 @@
+services:
+    espresso_database_creator:
+        image: postgres:16-alpine
+        command: ["createdb", "sequencer"]
+        depends_on:
+            database:
+                condition: service_healthy
+        environment:
+            PGHOST: ${PGHOST:-database}
+            PGPORT: ${PGPORT:-5432}
+            PGUSER: ${PGUSER:-postgres}
+            PGPASSWORD: ${PGPASSWORD:-password}
+            PGDATABASE: ${PGDATABASE:-postgres}
+
+    espresso:
+        image: ghcr.io/espressosystems/espresso-sequencer/espresso-dev-node:20240925
+        #image: espresso-dev-node:local
+        command: ["/usr/bin/espresso-dev-node"]
+        deploy:
+            resources:
+                limits:
+                    cpus: "4"
+                    memory: "1G"
+        ports:
+            - 8770:8770
+            - 8771:8771
+            - 8772:8772
+            - 20000:20000
+        depends_on:
+            espresso_database_creator:
+                condition: service_completed_successfully
+            database:
+                condition: service_healthy
+        environment:
+            ESPRESSO_SEQUENCER_L1_PROVIDER: ${CARTESI_BLOCKCHAIN_HTTP_ENDPOINT:-http://anvil:8545}
+            ESPRESSO_SEQUENCER_API_PORT: 8770
+            ESPRESSO_BUILDER_PORT: 8771
+            ESPRESSO_PROVER_PORT: 8772
+            ESPRESSO_DEV_NODE_PORT: 20000
+            ESPRESSO_SEQUENCER_POSTGRES_HOST: database
+            ESPRESSO_SEQUENCER_POSTGRES_PORT: 5432
+            ESPRESSO_SEQUENCER_POSTGRES_USER: postgres
+            ESPRESSO_SEQUENCER_POSTGRES_PASSWORD: password
+            ESPRESSO_SEQUENCER_POSTGRES_DATABASE: sequencer
+            ESPRESSO_SEQUENCER_ETH_MNEMONI: ${CARTESI_AUTH_MNEMONIC:-test test test test test test test test test test test junk}
+
+    prompt:
+        image: debian:bookworm-slim
+        environment:
+            PROMPT_TXT_07_ESPRESSO: "Espresso running at http://localhost:${CARTESI_LISTEN_PORT}/espresso/"
+
+    traefik-config-generator:
+        environment:
+            TRAEFIK_CONFIG_ESPRESSO: |
+                http:
+                    routers:
+                        espresso:
+                            rule: "PathPrefix(`/espresso`)"
+                            middlewares:
+                                - "remove-espresso-prefix"
+                            service: espresso
+                    middlewares:
+                        remove-espresso-prefix:
+                            replacePathRegex:
+                                regex: "^/espresso/(.*)"
+                                replacement: "/$1"
+                    services:
+                        espresso:
+                            loadBalancer:
+                                servers:
+                                    - url: "http://espresso:8770"
```

Comment:
there are other services, right?

Reply:
yeah, it's not clear yet if we need to expose any of these, and we'd need to add some kind of namespace for each port

like:

- `/espresso/builder/...` -> 8771
- `/espresso/prover/...` -> 8772
- ...

Reply:
I'm not sure either if we will need it, but if we don't add the prefix now we will have problems if we need them in the future.

Reply:
There's a fixup and you can use this mapping:

- `/espresso/dev` -> `espresso:20000`
- `/espresso/sequencer` -> `espresso:8770`
- `/espresso/builder` -> `espresso:8771`
- `/espresso/prover` -> `espresso:8772`

examples:

```shell
curl -fsSL http://localhost:8080espresso/dev/api/dev-info | jq .
{
  "builder_url": "http://localhost:8771/",
  "l1_prover_port": 8772,
  "l1_url": "http://anvil:8545/",
  "l1_light_client_address": "0x5f3f1dbd7b74c6b46e8c44f98792a1daf8d69154",
  "alt_chains": []
}
```

Reply:
This `builder_url`, if tried to be accessed from the host, will fail.
I don't know if this `/dev-info` is for information only, or is used by a service

Reply:
We'd need to know which endpoints to call and with which parameters (path).

https://docs.espressosys.com/sequencer/api-reference/builder-api

Reply:
We will mostly call APIs from the Sequencer API.

Reply:
Note that there are also WebSocket APIs, that we need to know if is supported by traefik

Reply:
I couldn't successfully  hit any ws endpoint without even using traefik.

AFAIK, traefik seamlessly handles the `HTTP` `Upgrade` to websocket, without any special configuration.